### PR TITLE
Added login page and functionality

### PR DESCRIPTION
**Pull request by rahulcs754**
- Added login page 
- added login functionality 
- stored token in local Storage
image

### CODE REVIEW

**Review by vaibhav-mahalle — APPROVED** [review]
you could improve your commit messages by following some kind of structure like feat,fix,chore, etc

File: src/frontend/Context/AuthReducer.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      localStorage.clear();
+      return {
+        ...state,
+        loading: false,
+        message: "",
+        isUserLoggedIn: false,
+        encodedToken: "",
+        userDetails: {},
+      };
+    case "LOGIN_ERROR":
```
Comment (coldpigli) [inline src/frontend/Context/AuthReducer.js:30]: Liked that you have reducers for loading and error states too

File: src/frontend/Context/AuthReducer.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      return {
+        ...state,
+        loading: false,
+        message: "",
+        isUserLoggedIn: true,
+        encodedToken: action.encodedToken,
+        userDetails: action.userDetails,
+      };
+    case "LOGOUT":
+      localStorage.clear();
```
Comment (coldpigli) [inline src/frontend/Context/AuthReducer.js]: you should only clear the local storage of this site. And not the other sites. 
So <code>localStorage.clear("tokenname)</code>

**Review by coldpigli — APPROVED** [review]
One suggestion. Everything else looks good!

File: src/frontend/Context/AuthReducer.js (hunk 2 above)
Comment (rahulcs754) [reply src/frontend/Context/AuthReducer.js]: Thanks for pointing out this. i can see changes in next commit

Comment (rahulcs754) [reply src/frontend/Context/AuthReducer.js]: done

**Pull request merged**